xds/internal/balancer/clusterresolver/resource_resolver_test.go

```diff
@@ -614,7 +615,7 @@ func (s) TestResourceResolverEDSAndDNS(t *testing.T) {
 	}
 	select {
 	case target := <-dnsTargetCh:
-		if diff := cmp.Diff(target, resolver.Target{Scheme: "dns", Endpoint: testDNSTarget}); diff != "" {
+		if diff := cmp.Diff(target, resolver.Target{Scheme: "dns", URL: *testutils.MustParseURL(testDNSTarget)}); diff != "" {
```

Comment:
@easwars, do we need to manually prepend scheme here (e.g., `"dns:///" + testDNSTarget`)? Any guidance as to when it would be necessary to do so and why it must be done manually?


FYI - there are a handful of lines in this file and another where I should potentially prepend scheme to target. Although not doing so does not seem to cause any errors in testing, I wanted to flag this inconsistency.

Reply:
I think `url.Parse()` doesn't like a URL without a scheme, because there is no default scheme as we have in grpc. So, wherever we need to pass a value directly to `url.Parse()`, the value would need to have a scheme.

Reply:
Ah, thanks! I went ahead and added scheme to remain consistent.

Let me know if you'd like me to revert that. Interestingly, tests do not appear to be affected by that decision.